### PR TITLE
MAINT: Sort `names.inc` using pre-commit hook

### PR DESCRIPTION
**Pull request by larsoner**
Should help us keep it sorted. Also gets in a CircleCI fix. Another option would have been to do
```
sort doc/changes/names.inc -o doc/changes/names.inc
```
in the `autofix.ci` job. But pre-commit hook seemed cleaner.

### CODE REVIEW

**Comment by larsoner** [comment]
Also going to push commits for this failing https://app.circleci.com/pipelines/github/mne-tools/mne-python/24815/workflows/55383920-e341-456d-b929-a8ee1c71169f/jobs/67690

File: doc/sphinxext/related_software.py [hunk 1]
```diff
@@ -24,7 +24,6 @@
 # If it's available on PyPI, add it to this set:
 PYPI_PACKAGES = {
     "alphaCSC",
-    "conpy",
```
Comment (larsoner) [inline doc/sphinxext/related_software.py:27]: We were linking to the wrong conpy (whoops!)

**Review by larsoner — COMMENTED** [review]
@drammock ready for review/merge from my end!

File: doc/conf.py [hunk 2]
```diff
@@ -121,6 +121,8 @@
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
 # This pattern also affects html_static_path and html_extra_path.
+
+# NB: changes here should also be made to the linkcheck target in the Makefile
```
Comment (drammock) [inline doc/conf.py:125]: Why do we want linkcheck to ignore our `_includes` folder? Isn't there valid rST in there that might contain links?

**Review by drammock — APPROVED** [review]
LGTM, just one question (hopefully not a dumb one; reviewing on a phone)

File: doc/conf.py (hunk 2 above)
Comment (larsoner) [reply doc/conf.py:125]: Yes but it gets `.. include::`ed somewhere else. If you don't exclude it then you get hundreds of warnings about duplicated link names

**Pull request merged**